gen/sam/sam_hawk.py

```diff
@@ -41,7 +41,7 @@ def generate(self):
         )
 
         # Triple A for close range defense
-        aa_group = self.add_auxiliary_group("AA")
+        aa_group = self.add_auxiliary_group(AirDefenseRange.AAA)
```

Comment:
I think this part of the refactor is good, but maybe it should be a new `AirDefenseGroupRole` instead. Both for current features and future flexibility, the range isn't what matters here; it's what skynet ought to do with it. I think the groups are (but some might be wrong and some might be missing):

* `RadarSAM`
* `EWR`
* `PointDefense`
* `AAA`
* `Support`